src/js/components/DataTable/DataTable.js

```diff
@@ -273,6 +273,40 @@ const DataTable = ({
       ? { style: { minWidth: '100%' } }
       : undefined;
 
+  const pinnedColumns = columns
+    .map(
+      (pinnedColumn, index) =>
+        pinnedColumn.pin && { property: pinnedColumn.property, id: index },
```

Comment:
Do we need to return `id` at all? I think we can just leverage `index` on line `289`

Reply:
We could also probably just return `pinnedColumn.property` and create an array of strings instead of an array of objects with just one key.